### PR TITLE
Allow the option to make field filter values required

### PR DESCRIPTION
**Pull request by cgopalan**
Currently metabase allows values filters to have a "Required" checkbox which the user can check on or off to make specifying a filter value required. This is not provided for field filters. This PR allows the field filters also to have the same option.

###### Before submitting the PR, please make sure you do the following 
-  [x] If there are changes to the backend codebase, run the backend tests with `lein test && lein eastwood && lein bikeshed && lein docstring-checker && ./bin/reflection-linter`
-  [x] Run the frontend and integration tests with  `yarn lint && yarn flow && yarn test`)
-  [x] Sign the [Contributor License Agreement](https://docs.google.com/a/metabase.com/forms/d/1oV38o7b9ONFSwuzwmERRMi9SYrhYeOrkbmNaq9pOJ_E/viewform)
(unless it's a tiny documentation change).

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/template_tags/TagEditorParam.jsx [hunk 1]
```diff
@@ -194,7 +194,7 @@ export default class TagEditorParam extends Component {
             </div>
           )}
 
-        {tag.type !== "dimension" && (
+	{(
```
Comment (tlrobinson) [inline frontend/src/metabase/query_builder/components/template_tags/TagEditorParam.jsx]: You can remove these brackets/parens if there's no conditional.

**Comment by mazameli** [comment]
Seems like this all works well, except that if you have the required toggle turned on, setting a default value doesn't seem to work to populate the filter:

screen shot 2018-10-16 at 1 34 52 pm

**Comment by cgopalan** [comment]
@mazameli so how should this behave? Setting the default filter widget value should automatically populate the filter above? I think after its saved, and the question is reopened again, the value does appear in the filter, just FYI.

**Comment by mazameli** [comment]
No, it doesn't need to populate the filter above. If you try turning the `Required` toggle off and entering a default value into the input box, and then run the query, that's the expected behavior. I.e., my expectation is that if `Required` is on, but there's a default value supplied, the query should run without me having to enter a value into the filter widget. Right now the query does not run. Hope that makes sense.

**Comment by cgopalan** [comment]
@mazameli Got it. Will look at this tomorrow.

**Comment by cgopalan** [comment]
@mazameli reminder on this :)

**Comment by mazameli** [comment]
Thanks for the ping! I'll take another look.

**Review by senior — COMMENTED** [review]
Looks good, just an indentation issue and a minor stylistic change.

**Comment by cgopalan** [comment]
@senior fixed. Please review.

**Comment by mazameli** [comment]
Looks good!

**Comment by mazameli** [comment]
Thanks for the contribution, @cgopalan!

**Comment by cgopalan** [comment]
@mazameli you are welcome. Love the product - and love working on your clojure codebase!

**Pull request merged**